### PR TITLE
update appnexus adapter to support new params

### PR DESCRIPTION
**Pull request by bokelley**
Support invcode+member
Improved testing framework
Support some new params (keywords, trafficSourceCode, reserve, position)

### CODE REVIEW

File: adapters/appnexus.go [hunk 1]
```diff
@@ -53,20 +65,54 @@ type appnexusImpExt struct {
...
+				// this assumes that the same member ID is used across all tags, which should be the case
+				uri = fmt.Sprintf("%s?member=%d", a.URI, params.Member)
+			}
+
+		}
+		if params.Reserve > 0 {
+			anReq.Imp[i].BidFloor = params.Reserve // TODO: we need to factor in currency here if non-USD
+		}
+		if anReq.Imp[i].Banner != nil && params.Position > 0 {
+			if params.Position == 1 {
```
Comment (mkendall07) [inline adapters/appnexus.go]: change to `above` and `below` as discussed.

File: adapters/appnexus_test.go [hunk 2]
```diff
@@ -1,241 +1,386 @@
...
+	defer r.Body.Close()
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
 	}
-}
 
-func TestAppNexusTimeout(t *testing.T) {
+	// fmt.Println("Request", string(body))
```
Comment (mkendall07) [inline adapters/appnexus_test.go]: remove unused code.

File: adapters/appnexus_test.go [hunk 3]
```diff
@@ -1,241 +1,386 @@
...
+	defer cancel()
+
+	bids, err = an.Call(ctx, pbReq, pbReq.Bidders[0])
+	if err == nil {
+		t.Fatalf("Should have gotten a timeout error: %v", err)
 	}
 }
 
-func TestAppNexusUserSyncInfo(t *testing.T) {
+/*
```
Comment (mkendall07) [inline adapters/appnexus_test.go]: drop instead of comment out?

File: adapters/appnexus_test.go [hunk 4]
```diff
@@ -1,241 +1,386 @@
...
+
+	body := new(bytes.Buffer)
+	err := json.NewEncoder(body).Encode(pbin)
+	if err != nil {
+		t.Fatalf("Json encoding failed: %v", err)
 	}
-}
 
-func TestAppNexusBasicResponse(t *testing.T) {
+	// fmt.Println("body", body)
```
Comment (mkendall07) [inline adapters/appnexus_test.go]: drop

File: adapters/appnexus.go [hunk 5]
```diff
@@ -37,14 +38,25 @@ func (a *AppNexusAdapter) GetUsersyncInfo() *pbs.UsersyncInfo {
...
+	Values []string `json:"value"`
+}
+
 type appnexusParams struct {
-	PlacementId int    `json:"placementId"`
-	InvCode     string `json:"invCode"`
-	Member      string `json:"member"`
+	PlacementId       int      `json:"placementId"`
+	InvCode           string   `json:"invCode"`
+	Member            int      `json:"member"`
```
Comment (mkendall07) [inline adapters/appnexus.go]: this is a breaking change...

Comment (bokelley) [inline adapters/appnexus.go]: not intended. Will fix.

File: adapters/appnexus.go (hunk 1 above)
Comment (bokelley) [inline adapters/appnexus.go]: will do

**Pull request merged**